sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusTestEnvironment.cs

```diff
@@ -135,6 +135,8 @@ public class ServiceBusTestEnvironment: TestEnvironment
         ///
         public string OverrideSecondaryServiceBusConnectionString => GetRecordedOptionalVariable(ServiceBusSecondaryConnectionStringEnvironmentVariable, options => options.HasSecretConnectionStringParameter("SharedAccessKey", SanitizedValue.Base64));
 
+        public string StorageClaimCheckConnectionString => GetRecordedVariable("STORAGE_CLAIM_CHECK_CONNECTION_STRING");
```

Comment:
This effectively breaks dynamic resource creation; we should consider either adding a `StorageTestEnvironment` and `StorageScope` similar to what Event Hubs uses _([ref](https://github.com/Azure/azure-sdk-for-net/tree/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobStorageTesting))_  or removing the functionality for dynamic Service Bus namespace creation to keep the SB/Storage needs for live tests consistent.

Reply:
Test resources created with New-TestResources will be deleted after 5 days by default. I think adding a StorageScope is a bit excessive for this one sample. I do need to update the test to delete the container though.

Reply:
That's my point.  :smile:

With this change, it is no longer possible to just run the tests if you have a persistent resource group without having to manually create/clean things.   The rest of the test infrastructure is self-managing.   If you don't have a namespace one is created when you run the tests and deleted after.   If you have one, it gets used and left in a clean state after the run.    

Now, you must create the storage account and if you want to avoid cost, you must clean it out.

I know that you feel differently about the dynamic resources, and I'm not pushing that....  just that we should have consistent behavior for them.

Reply:
I see your point, but I don't think it is actually self-managing - at least in its current state. If a namespace/account is deleted, but the env var is still set, it won't automatically create a new one for you.

Reply:
True, but it also won't delete it on you.   For example, I find it helpful to have both messaging services just pointed at an empty RG and run the tests.   Tests run,  I have an empty resource group again.  I'm not implying it's fool-proof, but it does enable the "just run your tests" scenario - which our current environment does not do with Test-Resources alone.

To be clear,  I'm not arguing that its essential we pull in the storage environment or even the scope.   My ask is that we just make a call on enabling the scenario or not.   I'm fine if you decide that Service Bus requires you to have a storage account and namespace created and specified an environment variable.   It just feels wrong to half-support it.

Reply:
_(it all becomes moot if we ever get F5 support for test resources, but so far that's not been a priority)_

Reply:
What do you mean by half-support it?

Reply:
> I'm fine if you decide that Service Bus requires you to have a storage account and namespace created and specified an environment variable.

This is what it is doing now.. I'm not sure I follow.

Reply:
Ah you mean remove it from test-resources.json

Reply:
> This is what it is doing now.. I'm not sure I follow.

Not quite.

I can not have a namespace specified and one gets dynamically created at the beginning of the run and then removed.   The same is not true for the storage account.   I must have one available and set in the env.

Reply:
Ah ok, since the functionality is already there for SB namespace, I agree it would be kind of annoying to not work for the Storage account. I will probably just steal the EH code here.